src/plugins/condition/components/inspector/StylesView.vue

```diff
@@ -0,0 +1,600 @@
+                        handleItemSelection
+                    }
+                },
+                template: '<condition-set-selector-dialog @conditionSetSelected="handleItemSelection"></condition-set-selector-dialog>'
+            }).$mount();
+
+            let overlay = this.openmct.overlays.overlay({
+                element: vm.$el,
+                size: 'small',
+                buttons: [
+                    {
+                        label: 'OK',
+                        emphasis: 'true',
+                        callback: () => dismissDialog(overlay, true)
+                    },
+                    {
+                        label: 'Cancel',
+                        callback: () => dismissDialog(overlay, false)
+                    }
+                ],
+                onDestroy: () => vm.$destroy()
+            });
+        },
+        removeConditionSet() {
+            this.conditionSetDomainObject = undefined;
+            this.conditionalStyles = [];
+            let domainObjectStyles =  (this.domainObject.configuration && this.domainObject.configuration.objectStyles) || {};
+            if (this.domainObjectsById) {
+                const domainObjects = Object.values(this.domainObjectsById);
+                domainObjects.forEach(domainObject => {
+                    let objectStyles =  (domainObject.configuration && domainObject.configuration.objectStyles) || {};
+                    this.removeConditionalStyles(objectStyles);
+                    if (_.isEmpty(objectStyles)) {
+                        objectStyles = undefined;
+                    }
+                    this.persist(domainObject, objectStyles);
+                });
+            }
+            if (this.items.length) {
+                this.items.forEach((item) => {
+                    const itemId = item.id;
+                    this.removeConditionalStyles(domainObjectStyles, itemId);
+                    if (_.isEmpty(domainObjectStyles[itemId])) {
+                        delete domainObjectStyles[itemId];
+                    }
+                });
+            }  else {
+                this.removeConditionalStyles(domainObjectStyles);
+            }
+            if (_.isEmpty(domainObjectStyles)) {
+                domainObjectStyles = undefined;
+            }
+            this.persist(this.domainObject, domainObjectStyles);
+
+            if (this.stopProvidingTelemetry) {
+                this.stopProvidingTelemetry();
+            }
+        },
+        removeConditionalStyles(domainObjectStyles, itemId) {
+            if (itemId && domainObjectStyles[itemId]) {
+                domainObjectStyles[itemId].conditionSetIdentifier = undefined;
+                delete domainObjectStyles[itemId].conditionSetIdentifier;
+                domainObjectStyles[itemId].selectedConditionId = undefined;
+                domainObjectStyles[itemId].defaultConditionId = undefined;
+                domainObjectStyles[itemId].styles = undefined;
```

Comment:
is there a reason why we are setting styles to undefined before we delete the variable?

Reply:
I can simplify it, but if the property doesn't exist on the object then a delete call on it will result in a not defined error.